### PR TITLE
feat(notifications): notifications 2.0 double write

### PR DESCRIPTION
**Pull request by scefali**
This PR writes to the new notification tables when the `organizations:notifications-double-write` flag is set. This way, after we run the migration we can ensure that the new tables are in sync with the old. Though we can turn the flag on before the migration to validate functionality first.

### CODE REVIEW

File: src/sentry/notifications/manager.py [hunk 1]
```diff
@@ -459,6 +564,32 @@ def update_settings_bulk(
...
+            # if default, delete the row
+            if value == NotificationSettingOptionValues.DEFAULT:
+                NotificationSettingOption.objects.filter(**query_args).delete()
+            else:
+                NotificationSettingOption.objects.create_or_update(
+                    **query_args,
+                    values={"value": NOTIFICATION_SETTING_OPTION_VALUES[value]},
+                )
+
+        # update the provider settings after we update other things
```
Comment (snigdhas) [inline src/sentry/notifications/manager.py:590]:
```suggestion
        # update the provider settings after we update the NotificationSettingOption
```

File: src/sentry/types/integrations.py [hunk 2]
```diff
@@ -49,6 +49,13 @@ class ExternalProviderEnum(Enum):
     ExternalProviders.CUSTOM: ExternalProviderEnum.CUSTOM.value,
 }
 
+# the list of providers allowed for personal notifications
+NOTIFICATION_PROVIDERS = [
```
Comment (snigdhas) [inline src/sentry/types/integrations.py:53]: Is this better named `PERSONAL_NOTIFICATION_PROVIDERS` for clarity?

File: src/sentry/notifications/helpers.py [hunk 3]
```diff
@@ -847,6 +851,32 @@ def user_has_any_provider_settings(
...
+        # done in a control silo so use the maping
+        org_ids = OrganizationMemberMapping.objects.filter(user_id=user_id).values_list(
+            "organization_id", flat=True
+        )
+        orgs = list(
+            map(
+                serialize_organization_mapping,
+                OrganizationMapping.objects.filter(organization_id__in=list(org_ids)),
+            )
+        )
```
Comment (markstory) [inline src/sentry/notifications/helpers.py:874]: Does this function run in both silos with different parameters? You won't be able to access teams in control silo.

Comment (scefali) [reply src/sentry/notifications/helpers.py:873]: @markstory i guess it has to run in the control silo, I'm getting around this by passing in `organization_id_for_team` so we can look up the feature flag that way.

**Review by snigdhas — APPROVED** [review]

**Pull request merged**